craft_application/application.py

```diff
@@ -236,7 +236,19 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
         """
         if project_dir is None:
             project_dir = self.project_dir
-        return (project_dir / f"{self.app.name}.yaml").resolve(strict=True)
+
+        project_file_path = project_dir / f"{self.app.name}.yaml"
+
+        try:
+            project_file = project_file_path.resolve(strict=True)
+        except FileNotFoundError as err:
+            raise errors.ProjectFileMissingError(
+                f"Project file '{str(project_file_path)}' not found.",
```

Comment:
What do you think about using just `f"{self.app.name}.yaml"` on the message? I think it should be clear enough, and for Snapcraft (again) there's no single place where the file could be.